HotFix/Core/Session.cs

```diff
@@ -342,6 +370,9 @@ public void InitiateLogon(IConfiguration configuration, State state, Channel cha
             throw new EngineException("Logon response not received on time");
         }
 
+        /// <summary>
+        /// Disconnects the underlying transport.
```

Comment:
Not sure if it's worth mentioning that `Logout` needs to be called one way or another before disposing for the session to end as close to properly as possible.